spec/controllers/profile_controller_spec.rb

```diff
@@ -16,5 +16,45 @@
       get :show
       expect(response.status).to eq(200)
     end
+
+    it "profile does not populate instance variables when in test" do
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      @user = FactoryBot.create(:user)
+      allow(request.env['warden']).to receive(:authenticate!).and_return(@user)
+      allow(controller).to receive(:current_user).and_return(@user)
+      sign_in @user
+      get :show
+      expect(@controller.instance_variable_get(:@stub_available)).to eq(nil)
+      expect(@controller.instance_variable_get(:@breakerofchains)).to eq(nil)
+      expect(@controller.instance_variable_get(:@using_stub)).to eq(nil)
+    end
+
+    it "profile populates instance variables when in production" do
+      Rails.env = 'production'
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      @user = FactoryBot.create(:user)
+      allow(request.env['warden']).to receive(:authenticate!).and_return(@user)
+      allow(controller).to receive(:current_user).and_return(@user)
+      sign_in @user
+      get :show
+      expect(@controller.instance_variable_get(:@stub_available)).to eq(nil)
+      expect(@controller.instance_variable_get(:@breakerofchains)).to eq(nil)
+      expect(@controller.instance_variable_get(:@using_stub)).to eq(nil)
+      Rails.env = 'test'
+    end
+
+    it "profile populates instance variables when in dev" do
+      Rails.env = 'development'
+      @request.env["devise.mapping"] = Devise.mappings[:user]
+      @user = FactoryBot.create(:user)
+      allow(request.env['warden']).to receive(:authenticate!).and_return(@user)
+      allow(controller).to receive(:current_user).and_return(@user)
+      sign_in @user
+      get :show
+      expect(@controller.instance_variable_get(:@stub_available)).to eq(true)
+      expect(@controller.instance_variable_get(:@breakerofchains)).to eq(false)
+      expect(@controller.instance_variable_get(:@using_stub)).to eq(true)
+      Rails.env = 'test'
+    end
```

Comment:
These can be  wrapped in their own context.

Possibly two separate contexts. If so, then `after_each` can be used to reset the Rails.env to test